aten/src/ATen/test/complex_test.cpp

```diff
@@ -0,0 +1,34 @@
+#include <gtest/gtest.h>
+#include "c10/util/Complex.h"
+
+template<typename T, typename iT>
+static void TestIntegerOp(T real, T img, iT i) {
+  std::complex<T> c(real, img);
+  ASSERT_EQ(c + i, std::complex<T>(real + i, img + i));
+  ASSERT_EQ(i + c, std::complex<T>(i + real, i + img));
+  ASSERT_EQ(c - i, std::complex<T>(real - i, img - i));
+  ASSERT_EQ(i - c, std::complex<T>(i - real, i - img));
+  ASSERT_EQ(c * i, std::complex<T>(real * i, img * i));
+  ASSERT_EQ(i * c, std::complex<T>(i * real, i * img));
+  ASSERT_EQ(c / i, std::complex<T>(real / i, img / i));
+  ASSERT_EQ(i / c, std::complex<T>(i / real, i / img));
+}
+
+template<typename Op>
+static void TestIntegerOpAllTypes(float real, float img, int8_t i) {
```

Comment:
this could be changed to:
```
template <typename T>
TestBinaryOpsForIntTypes(T real, T double, int num) {
TestBinaryOps<T, int8_t>(real, img, num);
TestBinaryOps<T, int16_t>(real, img, num);
TestBinaryOps<T, int32_t>(real, img, num);
TestBinaryOps<T, int64_t>(real, img, num);
}
```

Reply:
I removed template completely for this function (which isn't really useful).

Reply:
I think we should change the name of this function to TestBinaryOpsForIntTypes for more clarity